tests/stencils/test_sparse_fill.py

```diff
@@ -0,0 +1,75 @@
+from dusk.script import *
+from test_util import transpile_and_validate
+
+
+def test_sparse_fill():
+    transpile_and_validate(sparse_order_2_fill)
+    transpile_and_validate(longer_fills)
+    transpile_and_validate(fill_with_reduction)
+
+
+@stencil
+def sparse_order_2_fill(
+    vertex: Field[Vertex],
+    edge: Field[Edge, K],
+    cell: Field[Cell],
+    ve: Field[Vertex > Edge, K],
+    vc: Field[Vertex > Cell, K],
+    ev: Field[Edge > Vertex, K],
+    ec: Field[Edge > Cell, K],
+    cv: Field[Cell > Vertex, K],
+    ce: Field[Cell > Edge, K],
+):
+
+    with levels_upward:
+
+        with sparse[Vertex > Edge]:
+            ve = edge + 5
+        with sparse[Vertex > Cell]:
+            vc = cell - 6
+        with sparse[Edge > Vertex]:
+            ev = vertex * 7
+        with sparse[Edge > Cell]:
+            ec = cell / 8
+        with sparse[Cell > Vertex]:
+            cv = vertex ** 9
+        with sparse[Cell > Edge]:
+            ce = sqrt(edge)
+
+
+@stencil
+def longer_fills(
+    sparse1: Field[Vertex > Edge > Cell > Vertex > Edge > Cell],
+    sparse2: Field[Cell > Edge > Cell > Edge > Cell > Edge],
+    sparse3: Field[Edge > Cell > Vertex > Edge > Vertex > Edge],
+):
+    with levels_upward:
+        with sparse[Vertex > Edge > Cell > Vertex > Edge > Cell]:
+            sparse1 = 50
+
+    with levels_upward:
+        with sparse[Cell > Edge > Cell > Edge > Cell > Edge]:
+            sparse2 = 50
+
+    with levels_upward:
+        with sparse[Edge > Cell > Vertex > Edge > Vertex > Edge]:
+            sparse3 = 50
+
+
+@stencil
+def fill_with_reduction(
+    sparse1: Field[Edge > Cell > Vertex],
+    sparse2: Field[Edge > Cell > Vertex],
+    vertex: Field[Vertex],
+    edge: Field[Edge, K],
+    cell: Field[Cell],
+):
+    with levels_upward:
+        with sparse[Edge > Cell > Vertex]:
+            sparse1 = sum_over(Vertex > Cell, cell)
+            sparse2 = min_over(Vertex > Edge, edge)
+
+    with levels_upward:
+        with sparse[Edge > Cell > Vertex]:
+            sparse1 = sum_over(Vertex > Cell > Vertex, vertex[Vertex])
+            sparse2 = min_over(Vertex > Edge > Vertex, vertex[Vertex > Edge > Vertex])
```

Comment:
I see that all reductions here start from Vertex. Also having reductions that start from Edge should be valid syntax (not sure if it works though), since we allow accesses to edge fields. Could we try adding also those?

Reply:
I didn't know you could do that. And I'm not really sure how that makes sense.

```python
with sparse[Edge > Vertex]:
  sparse1 = sum_over(Edge > Cell, 1)
```
Which edge does the `sum_over` _start from_? The same as the `sparse[Edge > Vertex]`?

Would it be equivalent to?

```python
temp: Field[Edge]
temp = sum_over(Edge > Cell, 1)
with sparse[Edge > Vertex]:
  sparse1 = temp
```

Reply:
Yes to both questions.
In general, it makes sense to store data into a sparse field only when it depends on both
- some data on the starting location,
- some data on the target locations.
If one of the 2 can be excluded, you can express the same computation by using a dense field (either on the starting location type or on the target location type).
Data on the starting location may very well come from a reduction, as in your example with `ŧemp`.
By removing the temporary of your example (and computing it on-the-fly) you'd get valid code. However one could doubt whether it's correctly code generated.

Reply:
The ones that you added are the one that i mean, btw. 
Sorry for repeating myself, generated code looks ok also in this case?

Reply:
I didn't check the generated code. It's currently _outside the scope_ of dusk's testing infrastructure to properly verify the generated SIR. But it's definitely something I want to add in the future.